commit 1a9c9d919168a7f225cde88fccc3f9713b27f446
Author: Aaron Ma <aaron.ma@canonical.com>
Date:   Mon Oct 16 23:50:26 2017 +0800

    UBUNTU: SAUCE: Input: synaptics - add Intertouch support on X1 Carbon 6th and X280
    
    BugLink: http://bugs.launchpad.net/bugs/1723986
    
    Link: https://patchwork.kernel.org/patch/10008513/
    
    Synaptics devices reported it has Intertouch support,
    and it fails via PS/2 as following logs:
    
    psmouse serio2: Failed to reset mouse on synaptics-pt/serio0
    psmouse serio2: Failed to enable mouse on synaptics-pt/serio0
    
    Set these new devices to use SMBus to fix this issue,
    then they report SMBus version 3 is using,
    patch: https://patchwork.kernel.org/patch/9989547/ enabled SMBus ver 3 and
    makes synaptics devices work fine on SMBus mode.
    
    Signed-off-by: Aaron Ma <aaron.ma@canonical.com>
    Acked-by: Hui Wang <hui.wang@canonical.com>
    Acked-by: AceLan Kao <acelan.kao@canonical.com>
    Signed-off-by: Seth Forshee <seth.forshee@canonical.com>

diff --git a/drivers/input/mouse/synaptics.c b/drivers/input/mouse/synaptics.c
index b3c683a84d3f..12cdf18884d6 100644
--- a/drivers/input/mouse/synaptics.c
+++ b/drivers/input/mouse/synaptics.c
@@ -174,6 +174,8 @@ static const char * const smbus_pnp_ids[] = {
 	"LEN004a", /* W541 */
 	"LEN0073", /* X1 Carbon 5 */
 	"LEN200f", /* T450s */
+	"LEN0092", /* X1 Carbon 6 */
+	"LEN0096", /* X280 */
 	NULL
 };
 
